### PR TITLE
add support for Microsoft Teams

### PR DESCRIPTION
**Pull request by sleongkoan**

### CODE REVIEW

File: elastalert/alerts.py [hunk 1]
```diff
@@ -1295,3 +1295,36 @@ def alert(self, matches):
...
+        if isinstance(msteams_webhook_url, basestring):
+            msteams_webhook_url = [msteams_webhook_url]
+        self.msteams_webhook_url = msteams_webhook_url
+        self.msteams_proxy = self.rule.get('msteams_proxy')
+
+
+    def alert(self, matches):
+        payload = {
+            'isMultiline': True,
+            'text': '\n'.join(['```{}```'.format(match) for match in matches])
```
Comment (Qmando) [inline elastalert/alerts.py]: Almost every other alerter uses ``BasicMatchString(self.rule, match)`` to generate the alert text. It's fine to have the raw matches too, but I'd like to see that behind a config setting.

I know the other alerters are very inconsistent, but those need to be fixed, not copied.

**Comment by Qmando** [comment]
Can you also add some documentation?

File: elastalert/alerts.py (hunk 1 above)
Comment (sleongkoan) [inline elastalert/alerts.py]: Thanks for pointing that out. I now use `create_alert_body` to generate the body.

Documentation has also been updated

**Comment by sleongkoan** [comment]
@Qmando I think this PR is better than mine: https://github.com/Yelp/elastalert/pull/1130
you should have a look

**Comment by sleongkoan** [comment]
#1130 is better than this, closing this down

**Pull request closed without merging**